docs/chain-naming-standard.md

```diff
@@ -11,3 +11,18 @@ Where `network` is the value of the field that is written inside the
 [chain specification file](/docs/chain-specification-standard.md) that you are submitting.
 
 Please note that not using a valid name for the specification file will result into it being discarded.
+
+## Chain ID format
+
+A chain ID is a case-sensitive string in the format `^[a-zA-Z0-9_.-]{4,32}$`.
```

Comment:
I think this regex expression does not matches any case. Particularly, inside Cosmos: 

1. The network id is just a String, supporting even special characters. 
2. The network id is not limited in size 

I think that to be more generic and to be open for other chain specifications, this could ba changed removing the regex all together. 

```suggestion
```

Reply:
I don't think we can always copy chain IDs from cosmos into the registry 1:1. If we do that, we need namespacing in order to prevent a cosmos chain calling itself "eip155-123".

If we don't have a limit and a specified set of characters, we can never store any network ID from the registry in a blockchain, which is important for us.

Reply:
I see the point of having upper limits on the ID length. But the `eip155-123` is a valid chain id inside Cosmos, and it also satisfies all the specification you've written. 

How can we avoid such problems?

Reply:
Is there a chain ID specification in Cosmos that goes beyond the Tendermint genesis chain ID? In inter chain communication you need to have stricter rules as well, don't you?

Another point regarding special characters: if we leave a certain set of whitelisted characters, you cannot checkout this repository on Windows anymore since the net work ID is part of the file name.

Reply:
We will need to separate the chains between interfaces otherwise it will conflict with a Cosmos chain that has network `eip155-123`. Because they should both exist and we should be able to list them

Reply:
If we want to go the _we support them all_ path, I think we require namespacing by ecosystem. This could then look like

```
ethereum-eip155-123
cosmos-eip155-123
```

which also leads to

```
cosmos-cosmos-hub2
```

Reply:
I think this where we start defining our own standards, we can even do it externally as part of the Chain Agnostic Improvement Proposals (CAIPs) that @ligi created and then we can apply them here. In fact, that is actually the first CAIP that was created: ChainAgnostic/CAIPs#1

Regarding the name spacing per interface, I think we should use a colon (`:`) as a separator and then restrict network identifiers to not include them.

Reply:
I have added more comments to the CAIP-2 [here](https://github.com/ChainAgnostic/CAIPs/pull/1#issuecomment-534692931) and IMO we should close this PR and format our network ID as defined in my last example. CAIP-2 may or may not take longer and we could always revisit the network ID format later if necessary.

Reply:
Great stuff, let's go for that